modules/samples/doormanager/component/controller/src/main/java/org.homeautomation/doormanager/controller/api/DoorManagerControllerService.java

```diff
@@ -144,36 +147,36 @@ public void assignUseToLock(@HeaderParam("owner") String owner,
-                                response.setStatus(Response.Status.UNAUTHORIZED.getStatusCode());
-                                return;
-                            }
-                            Map<String, String> claims = new HashMap<>();
-                            claims.put("http://wso2.org/claims/lock/accesstoken", accessToken);
-                            claims.put("http://wso2.org/claims/lock/refreshtoken", accessTokenInfo.getRefresh_token());
-                            claims.put("http://wso2.org/claims/lock/cardnumber", cardNumber);
-                            userStoreManager.setUserClaimValues(userName, claims, null);
-                            //TODO: Add content to dto
-                        } else {
-                            response.setStatus(Response.Status.NOT_FOUND.getStatusCode());
+            if (userName != null && cardNumber != null && deviceId != null) {
+                try {
+                    UserStoreManager userStoreManager = this.getUserStoreManager();
+                    if (userStoreManager.isExistingUser(userName)) {
+                        TokenClient accessTokenClient = new TokenClient(DoorManagerConstants.DEVICE_TYPE);
+                        AccessTokenInfo accessTokenInfo = accessTokenClient.getAccessToken(deviceId, userName);
+                        String accessToken = accessTokenInfo.getAccess_token();
+                        if (accessToken == null) {
+                            response.setStatus(Response.Status.UNAUTHORIZED.getStatusCode());
+                            return;
                         }
-                    } catch (UserStoreException e) {
-                        response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
-                        log.error(e);
+                        Map<String, String> claims = new HashMap<>();
+                        claims.put("http://wso2.org/claims/lock/accesstoken", accessToken);
+                        claims.put("http://wso2.org/claims/lock/refreshtoken", accessTokenInfo.getRefresh_token());
+                        claims.put("http://wso2.org/claims/lock/cardnumber", cardNumber);
+                        userStoreManager.setUserClaimValues(userName, claims, null);
+                        //TODO: Add content to dto
+                        response.setStatus(Response.Status.OK.getStatusCode());
+                    } else {
+                        response.setStatus(Response.Status.NOT_FOUND.getStatusCode());
                     }
+                } catch (UserStoreException e) {
+                    response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+                    log.error(e);
                 }
-                else {
-                    response.setStatus(Response.Status.BAD_REQUEST.getStatusCode());
-                }
+            } else {
+                response.setStatus(Response.Status.BAD_REQUEST.getStatusCode());
```

Comment:
This only applicable for bad request. Why we should log bad request from client side in server side as an error?